File: Taller Tipos de clases/JAVA/Clase Abstracta/Animal.java
abstract class Animal {

    String nombre;

    // Constructor sin parámetros
    public Animal() {
        nombre = "Desconocido";
    }

    // Constructor con parámetros
    public Animal(String nombre) {
        this.nombre = nombre;
    }

    // Constructor copia
    public Animal(Animal a) {
        this.nombre = a.nombre;
    }

    // Método abstracto
    abstract void sonido();
}

File: Taller Tipos de clases/C#/Clase Abstracta/Animal.cs
abstract class Animal
{
    public string Nombre;

    public Animal()
    {
        Nombre = "Desconocido";
    }

    public Animal(string nombre)
    {
        Nombre = nombre;
    }

    public Animal(Animal a)
    {
        Nombre = a.Nombre;
    }

    public abstract void Sonido();
}


